### DISCUSSION TITLE
Wireless Keyball61 ?

### DISCUSSION THREAD

**Discussion by Elazul1122** [Yowkees/keyball#512] [Q&A]
Hi Yowkees,
hope you are doing well!
is it possible to make Keyball61 kit wireless by replacing the promicro with NRF52840 and Nice!nano bootloader?

**Comment by mybuild7** [comment]
Have you made an attempt? I hope Yowkees will answer, because I want to do a wireless build with the keyball 44. Maybe this may help you get an idea: "Charybdis Wireless Keyboard" on YouTube. It's a wireless trackball split keyboard build with a Nice!nano.

**Comment by KoichaDev** [comment]
Also interested to hear if someone managed to make the Keyball wireless

**Comment by KoichaDev** [comment]
I stumbled a Japanese website where someone is selling already pre-made keyball39 wireless.

https://bfgnvr.xuanfrph.top/index.php?main_page=product_info&products_id=282766&time=aHR0cHM6Ly9lbGRlc3ZhbmRlbGdhdG8uY29tL3Byb2R1Y3RzL3dvbmRlcmZ1bDgwMDI4Mjc2Nj9pZD0xMzg0NA==


Also, I see that on the description of the item, that there is someone that has forked the keyball39 and made their own firmware version that should work wirelessly. 

https://github.com/burakiyo/qmk_firmware/releases/tag/v1.0

The only issue is that I have absolutely no idea what type of micro controller to use and as well how to use the firmware above and following a procedure how to flash it. 


If someone maybe have an idea and like to try and make it work wirelessly, it would be really cool to get some feedback if it works or not

**Reply by elazul123** [reply]
> I stumbled a Japanese website where someone is selling already pre-made keyball39 wireless.
> 
> https://bfgnvr.xuanfrph.top/index.php?main_page=product_info&products_id=282766&time=aHR0cHM6Ly9lbGRlc3ZhbmRlbGdhdG8uY29tL3Byb2R1Y3RzL3dvbmRlcmZ1bDgwMDI4Mjc2Nj9pZD0xMzg0NA==
> 
> Also, I see that on the description of the item, that there is someone that has forked the keyball39 and made their own firmware version that should work wirelessly.
> 
> https://github.com/burakiyo/qmk_firmware/releases/tag/v1.0
> 
> The only issue is that I have absolutely no idea what type of micro controller to use and as well how to use the firmware above and following a procedure how to flash it.
> 
> If someone maybe have an idea and like to try and make it work wirelessly, it would be really cool to get some feedback if it works or not

many thanks mate this is really useful, I believe more of them will come!

**Comment by KoichaDev** [comment]
Another update!

I found another article written in Japanese:

- https://note.com/uk_oasis/n/nd59db23a7da5
- https://qiita.com/BlackChoco1ate-Xp10r/items/c23af9625b3a98b557ed
- https://qiita.com/tomcat09131/items/bda99cf68bf8da5c4729

You could use the Google Translate extension to translate it into English. 

I also stumbled on a repo that someone has forked from Yowkees and modified it to work Bluetooth. The only issue is that there is no guide on making the keyball61 build steps from the hardware perspective and making it compatible with the modified one forked one.

https://github.com/Code-Hex/ble-micro-pro-keyball61

Hopefully, someone will take the challenge and try it out to see if it works.

**Reply by chkpwd** [reply]
Any updates on this?

**Reply by AntoineGS** [reply]
I bought one on aliexpress and while it works it leaks 2mA due to the SPI being enable for the trackball. From what I can see a mosfet could resolve it but otherwise it works great with ZMK.
